### PR TITLE
Update integration tests console setup

### LINKED ISSUE #1518: Common console and login setup for integration tests
## Problem description

The integration tests are planned to be used for functional testing (boot, login, reboot)
Therefore they should all behave the same regarding the console and the login setup.
I suggest the following:

* bootloader console set to serial
* kernel console set to serial
* setup of sshd
* inclusion of an authorized_keys file to allow access without pwd

Security aspects are of no relevance for the integration testing. Nevertheless it might be
good to include ```/etc/motd``` to point out that this image description should not be used
for any production workload without access adaptions

### PR DESCRIPTION
**Pull request by schaefi**
Update all integration tests to use a serial tty console setup
and also to use a serial bootloader setup. This Fixes #1518

### CODE REVIEW

File: build-tests/arm/fedora/test-image-live/appliance.kiwi [hunk 1]
```diff
@@ -17,7 +17,7 @@
         <rpm-check-signatures>false</rpm-check-signatures>
     </preferences>
     <preferences>
-        <type image="iso" flags="overlay" firmware="uefi" hybridpersistent_filesystem="ext4" hybridpersistent="true"/>
+        <type image="iso" flags="overlay" firmware="uefi" hybridpersistent_filesystem="ext4" hybridpersistent="true" kernelcmdline="console=ttyS0"/>
```
Comment (davidcassany) [inline build-tests/arm/fedora/test-image-live/appliance.kiwi:20]: Shouldn't it also include the `<bootloader name="grub2" console="serial"/>`? This is leading to a grub2 installation based on templates, as far as I remember the templates also support this feature.

File: build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi [hunk 2]
```diff
@@ -31,23 +31,23 @@
         <rpm-check-signatures>false</rpm-check-signatures>
     </preferences>
     <preferences profiles="Live">
-        <type image="iso" flags="overlay" hybridpersistent_filesystem="ext4" hybridpersistent="true" firmware="efi"/>
+        <type image="iso" flags="overlay" hybridpersistent_filesystem="ext4" hybridpersistent="true" firmware="efi" kernelcmdline="console=ttyS0"/>
```
Comment (davidcassany) [inline build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi:34]: same here, I'd say the bootloader entry here makes sense too.

File: build-tests/x86/centos/test-image-live-disk-v7/appliance.kiwi [hunk 3]
```diff
@@ -25,20 +25,20 @@
     </preferences>
     <preferences profiles="Live">
         <type image="iso" flags="overlay" firmware="uefi" hybridpersistent_filesystem="ext4" hybridpersistent="true" kernelcmdline="console=ttyS0">
-            <bootloader name="grub2" console="serial gfxterm"/>
+            <bootloader name="grub2" console="serial"/>
```
Comment (davidcassany) [inline build-tests/x86/centos/test-image-live-disk-v7/appliance.kiwi:28]: See, this is a ISO with the bootloader entry.

File: build-tests/x86/suse/test-image-live/appliance.kiwi [hunk 4]
```diff
@@ -16,7 +16,7 @@
         <rpm-check-signatures>false</rpm-check-signatures>
         <bootsplash-theme>bgrt</bootsplash-theme>
         <bootloader-theme>openSUSE</bootloader-theme>
-        <type image="iso" flags="overlay" firmware="efi" kernelcmdline="console=ttyS0 splash" hybridpersistent_filesystem="ext4" hybridpersistent="true" mediacheck="true"/>
+        <type image="iso" flags="overlay" firmware="efi" kernelcmdline="console=ttyS0" hybridpersistent_filesystem="ext4" hybridpersistent="true" mediacheck="true"/>
```
Comment (davidcassany) [inline build-tests/x86/suse/test-image-live/appliance.kiwi:19]: Again the same, I'd probably include the bootloader entry here

File: build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi [hunk 5]
```diff
@@ -7,11 +7,11 @@
         <specification>SUSE Tumbleweed guest image for OpenStack</specification>
     </description>
     <preferences>
-        <type image="oem" filesystem="ext4" format="qcow2" kernelcmdline="console=tty1 console=ttyS0 net.ifnames=0">
+        <type image="oem" filesystem="ext4" format="qcow2" kernelcmdline="console=ttyS0 net.ifnames=0">
             <oemconfig>
                 <oem-resize>false</oem-resize>
             </oemconfig>
-            <bootloader name="grub2" timeout="1"/>
+            <bootloader name="grub2" timeout="1" console="serial gfxterm"/>
```
Comment (davidcassany) [inline build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi:14]: Any reason to keep the gfxterm here in this test and remove it on others?

**Review by davidcassany — COMMENTED** [review]
Just few comments regarding the usage of `bootloader` section.

File: build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi (hunk 5 above)
Comment (schaefi) [reply build-tests/x86/suse/test-image-qcow-openstack/appliance.kiwi:14]: yes to for testing the possible settings of the grub console parameter. I wanted to have at least one test that uses a multi console setup for grub to make sure the corresponding parts in the kiwi code are reached (e.g the pattern in the schema)

So this one is intentional

File: build-tests/x86/centos/test-image-live-disk-v7/appliance.kiwi (hunk 3 above)
Comment (schaefi) [reply build-tests/x86/centos/test-image-live-disk-v7/appliance.kiwi:28]: yep got it, I did the changes by substitution, so description with no bootloader setup felt of my radar

File: build-tests/x86/suse/test-image-live/appliance.kiwi (hunk 4 above)
Comment (schaefi) [reply build-tests/x86/suse/test-image-live/appliance.kiwi]: done

File: build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi (hunk 2 above)
Comment (schaefi) [reply build-tests/x86/archlinux/test-image-live-disk-kis/appliance.kiwi]: done

File: build-tests/arm/fedora/test-image-live/appliance.kiwi (hunk 1 above)
Comment (schaefi) [reply build-tests/arm/fedora/test-image-live/appliance.kiwi]: done

**Review by davidcassany — APPROVED** [review]
Looks good

**Pull request merged**